### PR TITLE
Chainmail standardization

### PR DESCRIPTION
**Pull request by Drew4484**
#### Summary
Content "Standardizes application of chainmail, removes chainmail/gambeson combos."

#### Purpose of change
As more items with chainmail are being updated, the standard of 1.2 mm (18 gauge) thickness will be used. After discussing chainmail, the decision was made to remove the integrated gambeson versions of chainmail armor. This allows greater player flexibility, removes a chainmail (outer) over chainmail gambeson (normal layer) exploit, and also cleans up repairing each item separately rather than using cotton to repair a damaged section of chainmail.

#### Describe the solution
Changes all standard chainmail to 1.2mm thickness, the same as 18 gauge steel.
Adjusts plate armors integrating chainmail thickness accordingly.
Removes gambesons from all chainmail suits/hauberks.
Obsoletes kevlar chainmail suits/hauberks.
Adds grades of steel to chainmails.

#### Describe alternatives you've considered
None.

#### Testing
Loaded items in game, verified changes.

#### Additional context
As the chainmail armors are having the gambesons removed, this may invalidate some of #55392.

### CODE REVIEW

**Comment by anoobindisguise** [comment]
1.2 mm may be the actual width of the chains themselves, but is it accurate? The chains are not perfectly flat, there is a somewhat uneven pattern to them that would increase thickness. This is going to give the chainmail a protection of only 6, which is low enough for common zombies to bleed you through full chainmail with claw swipes. and that doesn't seem reasonable considering that chainmail will generally protect you from full sword swipes. All steel armor suffers from a lack of cut protection but this seems particularly below real life.

**Comment by Drew4484** [comment]
Failing an unrelated test. @bombasticSlacks @Aerin-of-the-Toast if you have a minute to review before I ship this off, I'd appreciate it.

File: data/json/items/ammo.json [hunk 1]
```diff
@@ -1178,6 +1178,86 @@
     "ammo_type": "components",
     "count": 150
   },
+  {
+    "type": "AMMO",
+    "id": "lc_chain_link",
```
Comment (anothersimulacrum) [inline data/json/items/ammo.json:1183]: These links can probably all use copy-from.

File: data/json/items/generic.json [hunk 2]
```diff
@@ -698,6 +698,81 @@
     "volume": "100 ml",
     "to_hit": -2
   },
+  {
+    "type": "GENERIC",
+    "id": "lc_wire",
```
Comment (anothersimulacrum) [inline data/json/items/generic.json:703]: copy-from probably of use here too.

File: data/json/items/tool/metalworking.json [hunk 3]
```diff
@@ -254,6 +254,81 @@
     "color": "light_gray",
     "flags": [ "NO_SALVAGE" ]
   },
+  {
+    "id": "lc_link_sheet",
```
Comment (anothersimulacrum) [inline data/json/items/tool/metalworking.json:258]: And here.

File: data/json/items/ammo.json (hunk 1 above)
Comment (Drew4484) [reply data/json/items/ammo.json:1183]: Oh yep, I tried toc clean up with copy from but missed this one.

File: data/json/materials.json [hunk 4]
```diff
@@ -1642,6 +1642,116 @@
...
+  {
+    "type": "material",
+    "id": "lc_steel_chain",
+    "name": "Low Carbon Steel Chain",
+    "//": "SAE 1010/UNS 10100, assumes .1% carbon content and zero slag.",
+    "density": 8,
+    "specific_heat_liquid": 0.82,
+    "specific_heat_solid": 0.45,
+    "latent_heat": 273,
+    "bash_resist": 5,
```
Comment (anothersimulacrum) [inline data/json/materials.json:1654]: Should chain have a lower bash resist than equivalent steel plates? Subjectively, I would think so.

Comment (Drew4484) [reply data/json/materials.json:1654]: Possibly, but doing so compromises the plate armor that subtracts the chain thickness from the plate thickness. I'm not sure how else to reconcile chain and plate in a way that doesn't compromise the armor.

File: data/json/items/armor/arms_armor.json [hunk 5]
```diff
@@ -637,6 +643,180 @@
...
+    "description": "Customized mild steel chainmail arms.  Each sleeve has straps to connect them with each other.  The lack of fingers makes them less cumbersome and allows them to be used with gloves.  These are oversized.",
+    "proportional": { "weight": 1.25, "volume": 1.13, "price": 1.25 },
+    "extend": { "flags": [ "OVERSIZE" ] }
+  },
+  {
+    "id": "xs_lc_chainmail_arms",
+    "type": "ARMOR",
+    "copy-from": "lc_chainmail_arms",
+    "name": { "str": "pair of XS mild steel chainmail sleeves", "str_pl": "pairs of XS chainmail sleeves" },
+    "description": "Customized chainmail arms.  Each sleeve has straps to connect them with each other.  The lack of fingers makes them less cumbersome and allows them to be used with gloves.  These are undersized",
```
Comment (casswedson) [inline data/json/items/armor/arms_armor.json:684]:
```suggestion
    "description": "Customized mild steel chainmail arms.  Each sleeve has straps to connect them with each other.  The lack of fingers makes them less cumbersome and allows them to be used with gloves.  These are undersized.",
```

File: data/json/items/armor/arms_armor.json [hunk 6]
```diff
@@ -637,6 +643,180 @@
...
+    "description": "Customized medium steel chainmail arms.  Each sleeve has straps to connect them with each other.  The lack of fingers makes them less cumbersome and allows them to be used with gloves.  These are oversized.",
+    "proportional": { "weight": 1.25, "volume": 1.13, "price": 1.25 },
+    "extend": { "flags": [ "OVERSIZE" ] }
+  },
+  {
+    "id": "xs_mc_chainmail_arms",
+    "type": "ARMOR",
+    "copy-from": "mc_chainmail_arms",
+    "name": { "str": "pair of XS medium steel chainmail sleeves", "str_pl": "pairs of XS chainmail sleeves" },
+    "description": "Customized chainmail arms.  Each sleeve has straps to connect them with each other.  The lack of fingers makes them less cumbersome and allows them to be used with gloves.  These are undersized",
```
Comment (casswedson) [inline data/json/items/armor/arms_armor.json:717]:
```suggestion
    "description": "Customized medium steel chainmail arms.  Each sleeve has straps to connect them with each other.  The lack of fingers makes them less cumbersome and allows them to be used with gloves.  These are undersized.",
```

File: data/json/items/armor/arms_armor.json [hunk 7]
```diff
@@ -637,6 +643,180 @@
...
+    "description": "Customized high steel chainmail arms.  Each sleeve has straps to connect them with each other.  The lack of fingers makes them less cumbersome and allows them to be used with gloves.  These are oversized.",
+    "proportional": { "weight": 1.25, "volume": 1.13, "price": 1.25 },
+    "extend": { "flags": [ "OVERSIZE" ] }
+  },
+  {
+    "id": "xs_hc_chainmail_arms",
+    "type": "ARMOR",
+    "copy-from": "hc_chainmail_arms",
+    "name": { "str": "pair of XS high steel chainmail sleeves", "str_pl": "pairs of XS chainmail sleeves" },
+    "description": "Customized chainmail arms.  Each sleeve has straps to connect them with each other.  The lack of fingers makes them less cumbersome and allows them to be used with gloves.  These are undersized",
```
Comment (casswedson) [inline data/json/items/armor/arms_armor.json:750]:
```suggestion
    "description": "Customized high steel chainmail arms.  Each sleeve has straps to connect them with each other.  The lack of fingers makes them less cumbersome and allows them to be used with gloves.  These are undersized.",
```

File: data/json/items/armor/arms_armor.json [hunk 8]
```diff
@@ -637,6 +643,180 @@
...
+    "description": "Customized hardened steel chainmail arms.  Each sleeve has straps to connect them with each other.  The lack of fingers makes them less cumbersome and allows them to be used with gloves.  These are oversized.",
+    "proportional": { "weight": 1.25, "volume": 1.13, "price": 1.25 },
+    "extend": { "flags": [ "OVERSIZE" ] }
+  },
+  {
+    "id": "xs_ch_chainmail_arms",
+    "type": "ARMOR",
+    "copy-from": "ch_chainmail_arms",
+    "name": { "str": "pair of XS hardened steel chainmail sleeves", "str_pl": "pairs of XS chainmail sleeves" },
+    "description": "Customized chainmail arms.  Each sleeve has straps to connect them with each other.  The lack of fingers makes them less cumbersome and allows them to be used with gloves.  These are undersized",
```
Comment (casswedson) [inline data/json/items/armor/arms_armor.json:783]:
```suggestion
    "description": "Customized hardened steel chainmail arms.  Each sleeve has straps to connect them with each other.  The lack of fingers makes them less cumbersome and allows them to be used with gloves.  These are undersized.",
```

File: data/json/items/armor/arms_armor.json [hunk 9]
```diff
@@ -637,6 +643,180 @@
...
+    "description": "Customized tempered steel chainmail arms.  Each sleeve has straps to connect them with each other.  The lack of fingers makes them less cumbersome and allows them to be used with gloves.  These are oversized.",
+    "proportional": { "weight": 1.25, "volume": 1.13, "price": 1.25 },
+    "extend": { "flags": [ "OVERSIZE" ] }
+  },
+  {
+    "id": "xs_qt_chainmail_arms",
+    "type": "ARMOR",
+    "copy-from": "qt_chainmail_arms",
+    "name": { "str": "pair of XS tempered steel chainmail sleeves", "str_pl": "pairs of XS chainmail sleeves" },
+    "description": "Customized chainmail arms.  Each sleeve has straps to connect them with each other.  The lack of fingers makes them less cumbersome and allows them to be used with gloves.  These are undersized",
```
Comment (casswedson) [inline data/json/items/armor/arms_armor.json:816]:
```suggestion
    "description": "Customized tempered steel chainmail arms.  Each sleeve has straps to connect them with each other.  The lack of fingers makes them less cumbersome and allows them to be used with gloves.  These are undersized.",
```

Comment (Drew4484) [reply data/json/items/armor/arms_armor.json:816]: Thanks, thought I caught all of these.

**Pull request merged**